pyccel/codegen/printing/ccode.py

```diff
@@ -1876,7 +1876,10 @@ def _print_Allocate(self, expr):
             variable_address = self._print(ObjectAddress(expr.variable))
             container_type = self.get_c_type(expr.variable.class_type)
             if expr.alloc_type == 'reserve':
-                return free_code + f'{container_type}_reserve({variable_address}, {size});\n'
+                if expr.status == 'allocated':
+                    return (f'{container_type}_clear({variable_address});\n'
+                            f'{container_type}_reserve({variable_address}, {size});\n')
+                return f'{container_type}_reserve({variable_address}, {size});\n'
```

Comment:
```suggestion
                if expr.status != 'unallocated':
                    return (f'{container_type}_clear({variable_address});\n'
                            f'{container_type}_reserve({variable_address}, {size});\n')
                return f'{container_type}_reserve({variable_address}, {size});\n'
```
`clear` should also be called if the status is unknown (e.g maybe allocated in an if, maybe not)